src/components/ScreenWrapper.tsx

```diff
@@ -169,12 +169,18 @@ function ScreenWrapper(
     ).current;
 
     useEffect(() => {
+        // On iOS, the transitionEnd event doesn't trigger some times. As such, we need to set a timeout
+        const timeout = setTimeout(() => {
+            setDidScreenTransitionEnd(true);
+            onEntryTransitionEnd?.();
+        }, CONST.SCREEN_TRANSITION_END_TIMEOUT);
+
```

Comment:
@allroundexperts Is there any chance to avoid `setTimeout`? In which cases `transitionEnd` event doesn't trigger? Is this a known issue? May be we can solve the issue first? So, this code wouldn't be required.

cc: @mountiny

Reply:
As mentioned [here](https://github.com/Expensify/App/issues/41142#issuecomment-2081231869), this is a bug in react-native-navigation. For iOS, it doesn't seem to call the transition end handler.

Reply:
I think using `setTimeout` here isn't a bad practice. It's more of a safe exit.

Reply:
I think my worry is that we're changing the value of `didScreenTransitionEnd` using `setTimeout`.

Reply:
Right, so it is a known bug on iOS that sometimes `didScreenTransitionEnd` never gets set to `true` even though the transition is actually finished. Setting a `setTimeout` ensures that we handle such cases. Does that make sense @hayata-suenaga? Do you have any alternative that I should explore?

Reply:
I agree with you, @allroundexperts 😄. However, as we aim to avoid setTimeout as much as possible, I've asked in Slack if there are any other alternatives that don't involve using setTimeout. This is just to get a broader perspective before we decide to proceed with setTimeout.